ndbench-core/src/main/java/com/netflix/ndbench/core/util/ConstantStepWiseRateIncreaser.java

```diff
@@ -69,64 +60,26 @@ public ConstantStepWiseRateIncreaser(int rampPeriodMillisecs,
                     "rampPeriodMillisecs / incrementIntervalMillisecs should not exceed MAX_STEPS (" + MAX_STEPS + ")");
         }
 
-
-        final ArrayList<Cell> tmpClockTimeToRateTable = new ArrayList<Cell>();
-
         int numSteps = rampPeriodMillisecs / incrementIntervalMillisecs;
         double spread = (finalRate - initRate) * 1.0;
         double rateIncrementPerStep = spread / numSteps;
 
-        for (int i = 0; i <= numSteps; i++) {
-            tmpClockTimeToRateTable.add(
-                    new Cell(
-                            i * incrementIntervalMillisecs,
-                            initRate + (i * rateIncrementPerStep)
-                    )
-            );
-        }
-
-        // initialize high values sentinel cell
-        tmpClockTimeToRateTable.add(new Cell(Long.MAX_VALUE, 0 /* dummy never used for rate */));
-
-        indexOfLastInTable = numSteps + 1;
-        assert (tmpClockTimeToRateTable.size() == indexOfLastInTable + 1);
-        assert (tmpClockTimeToRateTable.get(numSteps).time == rampPeriodMillisecs);
-        assert (tmpClockTimeToRateTable.get(numSteps).rate == finalRate);
-
-
-        ImmutableList.Builder<Cell> cellBuilder = new ImmutableList.Builder<>();
-        clockTimeToRateTable = cellBuilder.addAll(tmpClockTimeToRateTable).build();
+        this.initRate = initRate;
+        this.finalRate = finalRate;
+        this.incrementIntervalMillisecs = incrementIntervalMillisecs;
+        this.rateIncrementPerStep = rateIncrementPerStep;
```

Comment:
Ditch `rateIncrementPerStep` local variable and just have `this.rateIncrementPerStep = spread / numSteps;`

Reply:
woops. ok